discounts/rust/order-discounts/fixed-amount/src/api.rs

```diff
@@ -0,0 +1,116 @@
+#![allow(dead_code)]
+
+use serde_with::{serde_as, DisplayFromStr};
+
+pub type Boolean = bool;
+pub type Decimal = f64;
+pub type Int = i32;
+pub type ID = String;
+
+pub mod input {
+    use super::*;
+    use serde::Deserialize;
+
+    #[serde_as]
+    #[derive(Clone, Debug, Deserialize)]
+    #[serde(rename_all(deserialize = "camelCase"))]
+    pub struct Input {
+        pub discount_node: DiscountNode,
+        pub presentment_currency_rate: Decimal,
+    }
+
+    #[derive(Clone, Debug, Deserialize, Default)]
+    pub struct DiscountNode {
+        pub metafield: Option<Metafield>,
+    }
+
+    #[derive(Clone, Debug, Deserialize, Default)]
+    pub struct Metafield {
+        pub value: String,
+    }
+}
+
+use serde::Serialize;
+use serde_with::skip_serializing_none;
+
+#[derive(Clone, Debug, Serialize)]
+#[serde(rename_all(serialize = "camelCase"))]
+pub struct FunctionResult {
+    pub discount_application_strategy: DiscountApplicationStrategy,
+    pub discounts: Vec<Discount>,
+}
+
+#[derive(Clone, Debug, Serialize)]
+#[serde(rename_all(serialize = "SCREAMING_SNAKE_CASE"))]
+pub enum DiscountApplicationStrategy {
+    First,
+    Maximum,
+}
+
+#[skip_serializing_none]
+#[derive(Clone, Debug, Serialize)]
+pub struct Discount {
+    pub value: Value,
+    pub targets: Vec<Target>,
+    pub message: Option<String>,
+    pub conditions: Option<Vec<Condition>>,
+}
+
+#[skip_serializing_none]
+#[serde_as]
+#[derive(Clone, Debug, Serialize)]
+#[serde(rename_all(serialize = "camelCase"))]
+pub enum Value {
+    FixedAmount {
+        #[serde_as(as = "DisplayFromStr")]
+        amount: Decimal,
+    },
+    Percentage {
+        #[serde_as(as = "DisplayFromStr")]
+        value: Decimal,
+    },
+}
+
+#[skip_serializing_none]
+#[derive(Clone, Debug, Serialize)]
+#[serde(rename_all(serialize = "camelCase"))]
+pub enum Target {
+    #[serde(rename_all(serialize = "camelCase"))]
+    OrderSubtotal {
+        excluded_variant_ids: Vec<ID>,
+    },
+    ProductVariant {
+        id: ID,
+        quantity: Option<Int>,
+    },
+}
+
+#[derive(Clone, Debug, Serialize)]
+#[serde(rename_all(serialize = "camelCase"))]
+pub enum Condition {
+    #[serde(rename_all(serialize = "camelCase"))]
+    OrderMinimumSubtotal {
+        excluded_variant_ids: Vec<ID>,
+        minimum_amount: Decimal,
```

Comment:
Do we need a `serde_as` here? (I think this will work but it's not what the docs say)